src/ui/HotkeyButton.tsx

```diff
@@ -33,20 +35,49 @@ export class HotkeyButton extends React.Component<Props, State> {
 
                         window.addEventListener("keypress", listener);
 
+                        const oldButtonState: boolean[][] = [];
+                        const intervalHandle = window.setInterval(() => {
+                            const gamepads = navigator.getGamepads();
+
+                            let gamepadIdx = 0;
+                            for (const gamepad of gamepads) {
+                                if (gamepad === null) { continue; }
+
+                                let buttonIdx = 0;
+                                for (const button of gamepad.buttons) {
+                                    const oldState = oldButtonState[gamepadIdx]?.[buttonIdx] ?? false;
+                                    if (button.pressed && !oldState) {
+                                        this.props.setValue(`Gamepad${buttonIdx}`);
+                                    }
+
+                                    if (oldButtonState[gamepadIdx] === undefined) {
```

Comment:
> Expression is always false.

No it's not, lmao. I love false positives on linters.

Reply:
Though this is kind of weird, because I feel like not even TypeScript itself models this properly. If you index out of bounds, there's no exception in JS and instead it's just `undefined`. However TypeScript doesn't understand that it seems. Probably for convenience reasons.